core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -254,13 +251,32 @@ oppia.controller('CollectionPlayer', [
       function(collectionBackendObject) {
         $scope.collection = CollectionObjectFactory.create(
           collectionBackendObject);
-        $scope.collectionPlaythrough = (
-          CollectionPlaythroughObjectFactory.create(
-            collectionBackendObject.playthrough_dict));
+
+        // Load the user's current progress in the collection. If the user is a
+        // guest, then either the defaults from the server will be used or the
+        // user's local progress, if any has been made.
+        if (!$scope.isLoggedIn &&
+            GuestCollectionProgressService.hasMadeProgress(
+              $scope.collectionId)) {
+          var completedExplorationIds = (
+            GuestCollectionProgressService.getCompletedExplorationIds(
+              $scope.collection));
+          var nextExplorationIds = (
+            GuestCollectionProgressService.getNextExplorationIds(
+              $scope.collection, completedExplorationIds));
+          $scope.collectionPlaythrough = (
+            CollectionPlaythroughObjectFactory.create(
+              nextExplorationIds, completedExplorationIds));
+        } else {
+          $scope.collectionPlaythrough = (
```

Comment:
Is there a playthrough dict sent to the frontend in the case where the user isn't logged in and hasn't made any progress? Even if so, would it be better to just do the following?

```
   if (!$scope.isLoggedIn) {
     if (no progress) create default
     else use existing progress
   } else {
     use backend progress
   }
```

Reply:
Yes, per ``summary_services.get_learner_collection_dict_by_id`` we always pass a playthrough dict to the frontend, and specialize it in the logged out case to the initial exp IDs with no current progress. That actually is the 'default' your snippet above refers to, and I think it's easier to rely on the backend to provide the correct defaults rather than recreating them locally (though we could via CollectionObjectFactory's getStartingCollectionNodes() or by reconstructing them via the guest services methods, it's duplicating logic with the backend).